### DISCUSSION TITLE
How to remove node_modules in all workspaces (nodeLinker node-modules)?

### DISCUSSION THREAD

**Discussion by jeffrson** [yarnpkg/berry#4094] [Q&A]
For a clean build with nodeLinker "node-modules" I want to remove existing node_modules in all workspaces. Apparently, this does not happen when running `yarn` from the root workspace.

I could use something like
`yarn workspaces foreach exec ...`,
however, this is dependent on successful `yarn install` in the workspaces' root (sensible yarn.lock needed). But I cannot/dont want to rely on that.

Another possibility would be to use "lerna clean -y", but I don't want to include lerna just for this.

What is the "recommended" way to that (other than writing platform dependent shell scripts myself)?

**Comment by arcanis** [comment]
> What is the "recommended" way to that (other than writing platform dependent shell scripts myself)?

You're not supposed to, `yarn install` will remove files that shouldn't be there on its own. If you really want to do it, `find . -name node_modules -exec rm -rf {} \;` will do the trick.

**Comment by jeffrson** [comment]
Thanks!

Unfortunately this did not work for me. After updating my mono-repository with git and running yarn I was facing problems related to left-overs of node_modules:
* old versions of libraries remaining
* lots of unneeded files remaining
Both situations could be resolved only by removing all node_modules prior to `yarn` (or `yarn install`). With Pnp this will be different, of course.

**Comment by Andre-3rd-Viking** [comment]
Found this answer:
- https://stackoverflow.com/a/70030644/704681
